core-processor/src/main/java/io/micronaut/aop/writer/AopProxyWriter.java

```diff
@@ -373,18 +444,8 @@ public boolean isProxyTarget() {
     }
 
     @Override
-    protected void startClass(ClassVisitor classWriter, String className, Type superType) {
-        String[] interfaces = getImplementedInterfaceInternalNames();
-        classWriter.visit(V17, ACC_SYNTHETIC, className, null, !isInterface ? superType.getInternalName() : null, interfaces);
```

Comment:
make sure that whatever we are generated retains `ACC_SYNTHETIC` modifier. I can't seem to see where this is retained?

Reply:
Ok, I need to add a support for it first